zerver/tests/test_decorators.py

```diff
@@ -108,6 +113,92 @@ def test(request, payload=REQ(argument_type="invalid")):
                 pass
             test(request)
 
+    def test_api_key_only_webhook_view(self):
+        @api_key_only_webhook_view('ClientName')
+        def get_user_profile_api_key(request, user_profile, client):
+            return user_profile.api_key
+
+        class Request(object):
+            REQUEST = {} # type: Dict[str, str]
+            COOKIES = {}
+            META = {'PATH_INFO': ''}
+
+        webhook_bot_email = 'webhook-bot@zulip.com'
+        request = Request()
+
+        request.REQUEST['api_key'] = 'not_existing_api_key'
+        with self.assertRaises(JsonableError):
+            get_user_profile_api_key(request)
+
+        request.REQUEST['api_key'] = get_user_profile_by_email(webhook_bot_email).api_key
+        request.COOKIES['csrftoken'] = get_token(request)
```

Comment:
I think you don't need to set `csrftoken` given that you're using API key auth...